### ISSUE TITLE
U2F NFC keys not working on Android  

### ISSUE THREAD

**Issue by mKoonrad** [brave/browser-android-tabs#1084]
Description:
I can't use U2F NFC keys (like YubiKey NEO) with Brave on Android. There is missing any FIDO U2F code/library in the Brave Android browser.

Steps to Reproduce:
1. Open Brave Android browser
2. Log in to any service with U2F support as 2 factor authentication like Google, GitHub etc.
3. Try to tap to phone NFC U2F key

Actual result:
Nothing. Page/service is waiting to use U2F key.

Expected result:
Tap to phone NFC U2F key and log in to service.

Issue reproduces how often:
Every time

Issue happens on:
current Playstore version and official build

Device Details:
phone OnePlus 5T
ARM version
Android 9

Brave version:
Brave 1.0.77, Chromium 71.0.3578.99

Website problems only:  
Issue doesn't resolve itself with disabling Brave Shields 
Issue isn't reproducible on the latest version of Google Chrome browser. On Brave you probably delete/not attach any code to support FIDO U2F protocol.

Additional Information
On Windows, PC version of Brave I don't have this issue. I have this issue only with Android.

**Comment by Rawi666** [comment]
Same issue here on OnePlus 6T (Android 9), Brave 1.0.89 / Chromium 73.0.3683.90.
This is quite annoying because if you have only U2F assigned to google account then there is no way you can log in through Brave at all.
However, logging through Chrome works (it uses Google Authenticator for U2F through NFC)

**Comment by Svarto** [comment]
Same here, Google pixel.

**Comment by drdownload** [comment]
Same here, pitty that it cant be used with my nfc u2f fido stick.

**Comment by Jonskar** [comment]
Still seeing the same issue here, hopefully it'll be prioritised soon.

**Comment by svengo** [comment]
Same here (Galaxy S9, Brave 1.0.95, Chromium 74.0.3729.157 on Android 9). Login with Google Chrome with the same version (74.0.3729.157) on the same phone works.

**Comment by aemacleod** [comment]
Same issue. Moto G7, Brave 1.0.95, Chromium 74.0.3729.157, Android 9. Chrome U2F works.

**Comment by callit** [comment]
Same - One Plus 7 Pro, Brave 1.0.95, Chromium 74.0.3729.157, Android 9

**Comment by dschuermann** [comment]
We could provide a FIDO implementation if the Brave team is interested: https://hwsecurity.dev/fido/

**Comment by mderazon** [comment]
Upstream Chromium already supports it, so isn't it a matter of catching up to latest Chromium ?

**Comment by dschuermann** [comment]
@mderazon On Android this feature is implemented by starting a authenticator Activity of the Google Play Services. Google Play Services are not part of the open source version. Thus, nope it is not available in the latest Chromium on Android.

**Comment by tomlowenthal** [comment]
Brave depends on Google Play Services in a bunch of ways, and that's not something we're currently looking to get out of.

**Comment by s-kk** [comment]
Is it possible to do a separate/dedicated implementation on brave instead of relying on Google's authentication service?

**Comment by vessokolev** [comment]
I have the same problem trying to access with secure key the Playground on https://demo.yubico.com. Seems like neither Google Chrome of Brave supports U2F NFC devices natively. I have tested 7 different Android devices so far, with different set of installed applications. No luck. So I can presume, there is very general problem with U2F that might not be entirely due to the lack of NFC or USB code in the open source version of Chromium. Note that, even in Google Chrome for Android, there is only experimental support for U2F NFC keys and U2F USB keys (it requires Google Authenticator, which is not updated since August 22, 2017!). I don't expect to see in the near future any mobile browser that supports U2F NFC / U2F USB devices natively.

**Comment by flipsa** [comment]
@vessokolev When I tested [https://demo.yubico.com](https://demo.yubico.com) with a friend's phone (Oneplus 5, latest OxygenOS, Chrome) it worked. But AFAIK Chrome is currently the only Android browser which supports U2F via NFC. On my custom ROM without GAPPS however, even Chrome fails, I guess because the Play-Store libraries aren't available.

**Comment by naftulikay** [comment]
Is there a tracking issue for USB? I can't even get my USB C YubiKey working on latest Android on Pixel 1.

**Comment by tomlowenthal** [comment]
U2F/WebAuthn is currently broken on Android, and the fix is part of a relatively-substantial refactor.

**Comment by srirambv** [comment]
Multiple +1 from https://community.brave.com/t/u2f-yubikey-nfc-not-working-in-brave-android/83929/5

**Comment by rlieving** [comment]
Not working on my Pixel 3 XL and it's really annoying. I love Brave on Android, but this lack of functionality breaks the browser in significant ways.

**Comment by srirambv** [comment]
+1 from https://community.brave.com/t/signing-in-to-google-with-fido2-key-mobile-browser/88530/2

**Comment by tomlowenthal** [comment]
Brave on Android is still mid-refactor, and it probably won't be finished until the end of the year. Until then, there aren't many options for fixing anything but the showstopperiest bugs.

**Comment by jphastings** [comment]
Is there any news, or way I can help with this? Google has been upping the marketing for their Titan Security Key in the UK this new year and I am now effectively locked out of my Google account on Brave as I can't authenticate using mine.

**Comment by tomlowenthal** [comment]
@jphastings No, there's nothing to be done here until the major refactor is shipping. @anthonypkeane & @SergeyZhukovsky should know the most about the timing for that.
